src/runtime_lib/infra_gapbs/graph.h

```diff
@@ -550,6 +563,37 @@ class CSRGraph {
   DestID_* get_in_neighbors_(void) {
       return in_neighbors_;
   }
+
+
+  inline int* get_flags_atomic_() {
+
+      static std::mutex thread_mutex;
```

Comment:
should make a comment here to note that there's only one instance of this static variables in the file or declare it globally.